[BUG] panic when passing --no-lb option and exposing port on the load balancer

## What did you do

- How was the cluster created?

```
k3d cluster create --no-lb --wait --image rancher/k3s:v1.17.14-k3s2 --port 80:80@loadbalancer --port 443:443@loadbalancer --port 27017:27017@server[0] --port 6379:6379@server[0] --port 9092:9092@server[0] --port 5432:5432@server[0] --k3s-server-arg "--disable=traefik"
```

- What did you do afterwards?
  - k3d commands?
N/A
  - docker commands?
N/A
  - OS operations (e.g. shutdown/reboot)?
N/A

## What did you expect to happen

I realise nobody should ever expose port on the load balancer when you specify `--no-lb` (I forgot to remove it when testing) but I think it shouldn't panic either when this situation happens.

## Screenshots or terminal output

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x126a615]

goroutine 1 [running]:
github.com/rancher/k3d/v3/cmd/cluster.parseCreateClusterCmd(0xc0000df8c0, 0xc000154120, 0x0, 0x12, 0xc000125020, 0xc00061fc30)
        github.com/rancher/k3d/v3/cmd/cluster/clusterCreate.go:438 +0x1a55
github.com/rancher/k3d/v3/cmd/cluster.NewCmdClusterCreate.func1(0xc0000df8c0, 0xc000154120, 0x0, 0x12)
        github.com/rancher/k3d/v3/cmd/cluster/clusterCreate.go:66 +0x94
github.com/spf13/cobra.(*Command).execute(0xc0000df8c0, 0xc000154000, 0x12, 0x12, 0xc0000df8c0, 0xc000154000)
        github.com/spf13/cobra@v1.1.0/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x1b69f20, 0xc000032170, 0x14, 0x15)
        github.com/spf13/cobra@v1.1.0/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.1.0/command.go:895
github.com/rancher/k3d/v3/cmd.Execute()
        github.com/rancher/k3d/v3/cmd/root.go:91 +0x5a
main.main()
        github.com/rancher/k3d/v3/main.go:27 +0x25
Error while creating the cluster
```

## Which OS & Architecture

- linux/amd64

## Which version of `k3d`

- v3.2.1

## Which version of docker

```
Client: Docker Engine - Community
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 17:02:36 2020
 OS/Arch:           linux/amd64
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

--- a/pkg/util/filter.go
+++ b/pkg/util/filter.go
@@ -86,6 +86,9 @@ func FilterNodes(nodes []*k3d.Node, filters []string) ([]*k3d.Node, error) {
 		} else if submatches["group"] == string(k3d.AgentRole) {
 			groupNodes = agentNodes
 		} else if submatches["group"] == string(k3d.LoadBalancerRole) {
+			if serverlb == nil {
+				return nil, fmt.Errorf("Node filter '%s' targets a node that does not exist (disabled?)", filter)
+			}
 			filteredNodes = append(filteredNodes, serverlb)
 			return filteredNodes, nil // early exit if filtered group is the loadbalancer
 		}

